src/m365/spo/commands/tenant/tenant-recyclebinitem-restore.spec.ts

```diff
@@ -66,83 +70,98 @@ describe(commands.TENANT_RECYCLEBINITEM_RESTORE, () => {
       }
 
       throw 'Invalid request';
     });
 
-    postRequestStub.onSecondCall().callsFake(async (opts) => {
-      if (opts.url === 'https://contoso-admin.sharepoint.com/_api/SPOInternalUseOnly.Tenant/RestoreDeletedSite') {
-        if (opts.headers && JSON.stringify(opts.data) === JSON.stringify({ siteUrl: 'https://contoso.sharepoint.com/sites/hr' })) {
-          return "{\"HasTimedout\":false,\"IsComplete\":false,\"PollingInterval\":100}";
-        }
+    await command.action(logger, { options: { siteUrl: siteUrl, verbose: true } });
+    assert.strictEqual(postStub.lastCall.args[0].url, groupRestoreUrl);
+  });
+
+  it('restores deleted site collection from the tenant recycle bin and does not remove m365 group when deleted site is not group connected', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === siteRestoreUrl) {
+        return;
       }
 
       throw 'Invalid request';
     });
 
-    postRequestStub.onThirdCall().callsFake(async (opts) => {
-      if (opts.url === 'https://contoso-admin.sharepoint.com/_api/SPOInternalUseOnly.Tenant/RestoreDeletedSite') {
-        if (opts.headers && JSON.stringify(opts.data) === JSON.stringify({ siteUrl: 'https://contoso.sharepoint.com/sites/hr' })) {
-          return "{\"HasTimedout\":false,\"IsComplete\":true,\"PollingInterval\":100}";
-        }
+    sinon.stub(odata, 'getAllItems').callsFake(async (url: string) => {
+      if (url === odataUrl) {
+        return [{ GroupId: '00000000-0000-0000-0000-000000000000' }];
       }
 
       throw 'Invalid request';
     });
 
-    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/hr', wait: true, verbose: true } });
-    assert(loggerLogSpy.calledWith({ HasTimedout: false, IsComplete: true, PollingInterval: 100 }));
+    await command.action(logger, { options: { siteUrl: siteUrl, verbose: true } });
+    assert(postStub.lastCall.args[0].url === siteRestoreUrl);
   });
 
   it('handles error when the site to restore is not found', async () => {
-    sinon.stub(request, 'post').callsFake(async (opts) => {
-      if ((opts.url as string).indexOf(`/_api/SPOInternalUseOnly.Tenant/RestoreDeletedSite`) > -1) {
-        if (opts.headers &&
-          JSON.stringify(opts.data) === JSON.stringify({
-            siteUrl: 'https://contoso.sharepoint.com/sites/hr'
-          })) {
-          throw "{\"odata.error\":{\"code\":\"-2147024809, System.ArgumentException\",\"message\":{\"lang\":\"en-US\",\"value\":\"Unable to find the deleted site: https://contoso.sharepoint.com/sites/hr.\"}}}";
+    const error = {
+      error: {
+        'odata.error': {
+          code: '-2147024809, System.ArgumentException',
+          message: {
+            lang: 'en-US',
+            value: `Unable to find the deleted site: ${siteUrl}`
+          }
         }
       }
+    };
+
+    sinon.stub(request, 'post').callsFake(async (opts) => {
```

Comment:
In this case, we don't really care if the API URL is correct, we only want to test that the command throws a clear error when there is an error.
So let's simplify this by using `sinon.stub(request, 'post').rejects(error);`